Reqnroll/EnvironmentAccess/BuildMetadataProvider.cs

```diff
@@ -0,0 +1,267 @@
+﻿using Reqnroll.CommonModels;
+
+namespace Reqnroll.EnvironmentAccess
+{
+    public class BuildMetadataProvider(IEnvironmentInfoProvider environmentInfoProvider, IEnvironmentWrapper environment) : IBuilldMetadataProvider
+    {
+        private string GetVariable(string variable)
+        {
+            var varValue = environment.GetEnvironmentVariable(variable);
+            if (varValue is ISuccess<string>)
+                return (varValue as ISuccess<string>).Result;
+            return null;
+        }
+        public BuildMetadata GetBuildMetadata()
+        {
+            var buildServer = environmentInfoProvider.GetBuildServerName();
+            return buildServer switch
+            {
+                "Azure Pipelines" => GetAzurePipelinesMetadata(),
```

Comment:
It's a bit strange in the architecture that we match on a string,  and the same string is returned in the method (product name)

Would be great if we could deduplicate that, e.g. by passing the buildserver variable